maskwatch-bot/__init__.py

```diff
@@ -88,18 +90,21 @@ async def _oper_up(self,
         else:
             await self.send(build("OPER", [oper_name, oper_pass]))
 
-    async def _is_oper(self, nickname: str):
+    async def _get_oper(self, nickname: str):
         await self.send(build("WHOIS", [nickname]))
 
-        whois_oper = Response(RPL_WHOISOPERATOR, [SELF, Folded(nickname)])
+        whois_oper = Response(RPL_WHOISSPECIAL, [SELF, Folded(nickname)])
         whois_end  = Response(RPL_ENDOFWHOIS,    [SELF, Folded(nickname)])
-        #:lithium.libera.chat 313 sandcat sandcat :is an IRC Operator
+        #:lithium.libera.chat 320 sandcat sandcat :is opered as sandcat, privset sandcat
         #:lithium.libera.chat 318 sandcat sandcat :End of /WHOIS list.
 
         whois_line = await self.wait_for({
             whois_end, whois_oper
         })
-        return whois_line.command == RPL_WHOISOPERATOR
+        # return the oper name or nothing
+        if whois_line.command == RPL_WHOISSPECIAL and whois_line.params[2].startswith("is opered as"):
+            return whois_line.params[2].split(",")[0].split()[-1]
```

Comment:
you won't always be told the privset, so sometimes you'll get
```
is opered as sandcat, privset sandcat
```
and sometimes you'll get
```
is opered as sandcat
```

so I'd instead suggest this
```python3
if whois_line.command == RPL_WHOISOPERATOR:
    match = re.search(r"^is opered as (\S+)(?:,|$)", whois_line.params[2])
    if match is not None:
        return match.group(1)
```